src/services/weight.ts

```diff
@@ -3,14 +3,15 @@ import { WeightAdapter, WeightEntry } from "components/BodyWeight/model";
 import { ApiBodyWeightType } from 'types';
 import { makeHeader, makeUrl } from "utils/url";
 import { ResponseType } from "./responseType";
+import { FilterType } from '../components/BodyWeight/widgets/FilterButtons';
 
 export const WEIGHT_PATH = 'weightentry';
 
 /*
- * Fetch all weight entries
+ * Fetch weight entries based on filter value
  */
-export const getWeights = async (): Promise<WeightEntry[]> => {
-    const url = makeUrl(WEIGHT_PATH, { query: { ordering: '-date', limit: 900 } });
+export const getWeights = async (filter: FilterType = ''): Promise<WeightEntry[]> => {
+    const url = makeUrl(WEIGHT_PATH, { query: { ordering: '-date', limit: 900, filter } });
```

Comment:
sorry I forgot one more! The backend doesn't currently understand the filter parameter, but we can calculate the date based on it and pass it like this `'date_gte': <iso-date-string>`

Reply:
I have also modified the backend to support the filter and linked the pull request to the issue, alongside this one. Would you prefer the solution I provided, or would you like me to adjust it according to your suggestion and leave the backend unchanged?

Reply:
I saw that after I wrote this 😄. I think putting the specific date is better since it's more explicit and we don't need a special syntax for that (you'll need to pull master for this BTW)